nipoppy/zenodo.py

```diff
@@ -0,0 +1,70 @@
+"""Client for Zenodo API."""
+
+import hashlib
+from pathlib import Path
+
+import httpx
+
+
+class InvalidChecksumError(Exception):
+    """Checksum mismatch between two files."""
+
+    pass
+
+
+class ZenodoAPI:
+    """Client to interact with Zenodo API.
+
+    Zenodo uses the InvenioRDM API, which is documented at:
+    https://inveniordm.docs.cern.ch/reference/rest_api_index/
+    """
+
+    def __init__(self, api_endpoint: str, access_token: str):
+        self.api_endpoint = api_endpoint
+        self.access_token = access_token
+        self.headers = {
+            "Authorization": f"Bearer {self.access_token}",
+        }
+
+    def download_record_files(self, zenodo_id: str, output_dir: Path):
+        """Download the files of a Zenodo record in the `output_dir` directory.
+
+        Parameters
+        ----------
+        zenodo_id : str
+            Record ID in Zenodo.
+        output_dir : Path
+            Output directory to save the files.
+
+        Raises
+        ------
+        InvalidChecksumError
+            Checksum mismatch between the downloaded file and the expected checksum.
+        """
+        output_dir.mkdir(parents=True, exist_ok=True)
+        response = httpx.get(
+            f"{self.api_endpoint}/records/{zenodo_id}/files",
+            headers=self.headers,
+        )
+        if response.status_code != 200:
+            raise Exception(response.json())
+
+        # Exclude "md5:" prefix
+        files = {
+            entry["key"]: entry["checksum"][4:] for entry in response.json()["entries"]
```

Comment:
I think using `removeprefix()` is cleaner:
```suggestion
            entry["key"]: entry["checksum"].removeprefix("md5:") for entry in response.json()["entries"]
```